src/azureFunction/azureFunctionUtils.ts

```diff
@@ -0,0 +1,154 @@
+	const settings: ILocalSettingsJson = await getLocalSettingsJson(localSettingsPath);
+
+	settings.Values = settings.Values || {};
+	if (settings.Values[key] === value) {
+		// don't do anything if it's the same as the existing value
+		return true;
+	} else if (settings.Values[key]) {
+		const result = await vscode.window.showWarningMessage(constants.settingAlreadyExists(key), { modal: true }, constants.yesString);
+		if (result !== constants.yesString) {
+			// key already exists and user doesn't want to overwrite it
+			return false;
+		}
+	}
+
+	settings.Values[key] = value;
+	await fse.writeJson(localSettingsPath, settings, { spaces: 2 });
+
+	return true;
+}
+
+/**
+ * Adds specified package to Azure Functions project the specified file is a part of
+ * @param filePath uri of file to find the containing AF project of to add package reference to
+ * @param packageName package to add reference to
+ * @param packageVersion optional version of package. If none, latest will be pulled in
+ */
+export async function addPackageToAFProjectContainingFile(
+	fileUri: vscode.Uri, packageName: string,
+	packageVersion?: string): Promise<void> {
+	try {
+		const project = await getAFProjectContainingFile(fileUri);
+
+		// if no AF projects were found, an error gets thrown from getAFProjectContainingFile(). This check is temporary until
+		// multiple AF projects in the workspace is handled. That scenario returns undefined and shows an info message telling the
+		// user to make sure their project has the package reference
+		if (project) {
+			await this.addPackage(project, packageName, packageVersion);
+		} else {
+			void vscode.window.showInformationMessage(constants.addPackageReferenceMessage, constants.moreInformation).then((result) => {
+				if (result === constants.moreInformation) {
+					void vscode.env.openExternal(vscode.Uri.parse(constants.sqlBindingsHelpLink));
+				}
+			});
+		}
+	} catch (e) {
+		const result = await vscode.window.showErrorMessage(constants.addSqlBindingPackageError, constants.checkoutOutputMessage);
+		if (result === constants.checkoutOutputMessage) {
+			this._outputChannel.show();
+		}
+	}
+}
+
+
+/**
+ * Gets the Azure Functions project that contains the given file if the project is open in one of the workspace folders
+ * @param filePath file that the containing project needs to be found for
+ * @returns uri of project or undefined if project couldn't be found
+ */
+export async function getAFProjectContainingFile(fileUri: vscode.Uri): Promise<vscode.Uri | undefined> {
+	// get functions csprojs in the workspace
+	const projectPromises = vscode.workspace.workspaceFolders?.map(f => getAllProjectsInFolder(f.uri, '.csproj')) ?? [];
+	const functionsProjects = (await Promise.all(projectPromises)).reduce((prev, curr) =>
+		prev.concat(curr), []).filter(p => isFunctionProject(path.dirname(p.fsPath)));
+
+	// look for project folder containing file if there's more than one
+	if (functionsProjects.length > 1) {
```

Comment:
Are we going to warn the user when this happens?

Reply:
As of this moment it seems sql-database-projects also just console.error out at the moment. But I can put something like this for now - 

> 		vscode.window.showWarningMessage('Unable to find which project contains the file: ' + fileUri.fsPath);